src/core/dbus-manager.c

```diff
@@ -2937,6 +2937,168 @@ static int method_dump_unit_descriptor_store(sd_bus_message *message, void *user
         return method_generic_unit_operation(message, userdata, error, bus_service_method_dump_file_descriptor_store, 0);
 }
 
+static int aux_scope_from_message(Manager *m, sd_bus_message *message, Unit **ret_scope, sd_bus_error *error) {
+        _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
+        _cleanup_free_ pid_t *pids = NULL;
+        const char *name, *suffix;
+        Unit *from, *scope;
+        CGroupContext *cc;
+        size_t n_pids = 0;
+        pid_t pid, main;
+        uint64_t flags;
+        int r;
+
+        assert(ret_scope);
+
+        r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID, &creds);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_creds_get_pid(creds, &pid);
+        if (r < 0)
+                return r;
+
+        from = manager_get_unit_by_pid(m, pid);
```

Comment:
hmm, we should make a stricter check here, only allow for processes from the source service to be "split out" like this, not arbitrary processes

Reply:
This is already done bellow.